mvcc/kvstore.go

```diff
@@ -160,6 +166,56 @@ func (s *store) Hash() (hash uint32, revision int64, err error) {
 	return h, s.currentRev, err
 }
 
+func (s *store) HashByRev(rev int64) (hash uint32, currentRev int64, compactRev int64, err error) {
+	readTx := s.b.ReadTx()
+	for {
```

Comment:
this isn't needed with readtx, it should be enough to have TxnRead:
kvstore_txn.go:
```go
func (s *store) Read() TxnRead { return newMetricsTxnRead(s.read()) }

func (s *store) read() *storeTxnRead {
        s.mu.RLock()
        tx := s.b.ReadTx()
        s.revMu.RLock()
        tx.Lock()
        firstRev, rev := s.compactMainRev, s.currentRev
        s.revMu.RUnlock()
        return &storeTxnRead{s, tx, firstRev, rev}
}
```

```go
func (s *store) HashByRev(rev int64) (int32, int64, int64) {
    txnRead := s.read()
    defer txnRead.End()
    ...
    txnRead.tx.UnsafeForEach(keyBucketName, func(k, v []byte) error {
        ...
    }
    ...
    return hash, readTxn.Rev(), readTxn.FirstRev()
}
```